shared/p2p/service_test.go

```diff
@@ -70,15 +71,74 @@ func TestP2pPortTakenError(t *testing.T) {
 }
 
 func TestBroadcast(t *testing.T) {
-	s, err := NewServer(&ServerConfig{})
+	s, err := NewServer(&ServerConfig{Port: 12345})
+	if err != nil {
+		t.Fatalf("error while trying to create server: %s", err)
+	}
+
+	s2, err := NewServer(&ServerConfig{Port: 54321})
+	if err != nil {
+		t.Fatalf("error while trying to create server: %s", err)
+	}
+
+	s3, err := NewServer(&ServerConfig{Port: 11223})
 	if err != nil {
-		t.Fatalf("Could not start a new server: %v", err)
+		t.Fatalf("error while trying to create server: %s", err)
+	}
+
+	s.Start()
+	s2.Start()
+	s3.Start()
+
+	time.Sleep(time.Second * 5)
+
+	for _, peer := range s.host.Peerstore().Peers() {
```

Comment:
How did the peerstore get populated?

Shouldn't you do this:
```go
s.host.Connect(ctx, peerstore.PeerInfo{ID: s1.host.ID(), Addrs: s.host.Addrs()})
```

See this example: https://github.com/libp2p/go-libp2p-examples/blob/1992b4ce7e838b5409437e5931d5abcf2b05f530/relay/main.go#L40

If you are relying on mdns discovery, I really wouldn't do that as it's going to make this test very flaky.